versioned/persist/tx/src/main/java/org/projectnessie/versioned/persist/tx/h2/H2DatabaseAdapter.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.versioned.persist.tx.h2;
+
+import java.util.Arrays;
+import java.util.List;
+import org.projectnessie.versioned.persist.tx.TxDatabaseAdapter;
+import org.projectnessie.versioned.persist.tx.TxDatabaseAdapterConfig;
+
+public class H2DatabaseAdapter extends TxDatabaseAdapter {
+
+  public H2DatabaseAdapter(TxDatabaseAdapterConfig config) {
+    super(config);
+  }
+
+  @Override
+  protected List<String> databaseSqlFormatParameters() {
+    return Arrays.asList(
```

Comment:
should this have been a map or something more typed?

Reply:
These parameters are (only) used to generate "rudimentary" `CREATE TABLE` DDLs (think: the DDLs work, but are probably not optimized for a particular database). Those DDLs are "formatted" with the data types from this method using `MessageFormat`, which unfortunately only allows parameter-indexes but not parameter-names.

Reply:
Could we have an `enum` with a custom field for the formatting index?

Reply:
Could. But what if another implementation needs more parameters, because it overrides `org.projectnessie.versioned.persist.tx.TxDatabaseAdapter#allCreateTableDDL`?

The combination of `allCreateTableDDL` + `databaseSqlFormatParameters` isn't great, but at the moment it limits the amount of rdbms-specific code to a minimum. I suspect that "proper setups" will require "highly optimized / customized" DDL scripts.

Reply:
I think this is pretty icky. I don't know what the best solution is but currently this is rather smelly and error prone.

Reply:
updated this a bit

Reply:
The update impl looks reasonable to me. As for me I only have one minor concern about the `SqlDataType` enum name - added a separate comment for that.